### ISSUE TITLE
[BUG] When salt is installed via pip for python3 separate from system python3, State 'virt.keys' was not found in SLS 'libvirt.server.tls.certs'

### ISSUE THREAD

**Issue by allyunion** [saltstack-formulas/libvirt-formula#76]
### Formula commit hash / release tag

v3.7.5

### Versions reports (master & minion)

Master:
```
Salt Version:
           Salt: 3000.2

Dependency Versions:
           cffi: 1.14.0
       cherrypy: unknown
       dateutil: 2.6.1
      docker-py: 1.10.6
          gitdb: 4.0.4
      gitpython: 3.1.1
         Jinja2: 2.11.2
        libgit2: 1.0.0
       M2Crypto: 0.35.2
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: 1.4.6
      pycparser: 2.20
       pycrypto: 3.9.7
   pycryptodome: Not Installed
         pygit2: 1.2.0
         Python: 3.7.5 (default, Nov  7 2019, 10:50:52)
   python-gnupg: 0.4.6
         PyYAML: 3.12
          PyZMQ: 19.0.0
          smmap: 3.0.2
        timelib: 0.2.4
        Tornado: 4.5.3
            ZMQ: 4.3.2

System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-99-generic
         system: Linux
        version: Ubuntu 18.04 bionic
```

Minion:

```
Salt Version:
           Salt: 3000.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.11.2
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.7.7 (default, May  4 2020, 05:28:05)
   python-gnupg: Not Installed
         PyYAML: 5.3.1
          PyZMQ: 19.0.0
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.2

System Versions:
           dist: debian bullseye/sid
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-96-generic
         system: Linux
        version: debian bullseye/sid
```

### Pillar / config used

/etc/salt/master:
file_roots:
  base:
    - /srv/salt
    - /srv/formulas/libvirt-formula

/srv/salt/top.sls:
base:
  '*':
    - libvirt

---

### Describe the bug

```
          ID: libvirt-server-tls-certs-virt-keys
    Function: virt.keys
        Name: libvirt_keys
      Result: False
     Comment: State 'virt.keys' was not found in SLS 'libvirt.server.tls.certs'
              Reason: 'virt' __virtual__ returned False
     Changes:
```

### Steps to reproduce the bug

1. Install salt via package on Ubuntu
1. Install Python 3.7.7 manually (Ubuntu 20.04 LTS)
1. Install salt to latest via pip using Python 3.7.7
1. Link salt binaries from /usr/local/bin to /usr/bin
1. Perform config file changes for master/minion
1. Enable daemons in Ubuntu
1. Run high state

### Expected behaviour

This error does not show up

### Attempts to fix the bug

1. On the minion, install libvirt-dev package
1. Then install libvirt-python via pip in the same Python interpreter used for Salt.

**Comment by baby-gnu** [comment]
Hello.

Thanks for the report, in the steps to reproduce, is there any reason you install salt via package on Ubuntu?

The formula only manage packages from the distribution and does not support PIP installation. It looks like the `python3-package` installed from `apt` can't be used by Python3.7 installed manually.

**Comment by myii** [comment]
If I understand correctly, this is because Salt isn't available on Ubuntu 20.04 until Sodium (mid-June) because the OS ships with Python 3.8.  So this may end up being quite tricky to resolve.  I think I've heard that some people have managed to get Salt installed on Python 3.8 by using the Debian packages in some way, @allyunion you might want to ask in Slack about that.

**Comment by javierbertoli** [comment]
If it's worth the comment, I installed salt on a Debian 'testing' host (with python 3.8) using packages and it didn't work for me. Although installation was trivial and proceeded with no errors, master was failing to retrieve and build pillars for the minions, with  some `os_family` grains errors in my setup.

Simply reverting to 'stable' (python 3.7) versions of the python packages and removing all python 3.8 made it work again, with no change in my salt config, states or pillars.

Sadly I don't have that setup around (destroyed the instance a few days ago) to paste actual errors, but just wanted to mention them so you can take the situation into consideration, as Ubuntu is based on Debian unstable and I guess the errors you face with it might be similar to those I found?

Hope this helps.

**Comment by allyunion** [comment]
@myii : There is no support for Salt for Python 3.8 and won't be for awhile either, I don't think.

**Comment by myii** [comment]
@allyunion I've seen some discussions about how to get this work, such as this comment:

* https://github.com/saltstack/salt/issues/56798#issuecomment-617540294

There were also some discussions on Slack.

But there's no official support until Sodium (3001).

**Comment by allyunion** [comment]
Okay, I'm now running Sodium (3001) and the error still shows.

**Comment by didiermfb** [comment]
As I use onedir package I also have this issue.
To resolve this on debian, I have to install libvirt-dev and pkg-config system packages and libvirt-python in the salt python virtual environment (salt-pip)
Those packages should be required in the formula.

**Comment by baby-gnu** [comment]
> As I use onedir package I also have this issue. To resolve this on debian, I have to install libvirt-dev and pkg-config system packages and libvirt-python in the salt python virtual environment (salt-pip) Those packages should be required in the formula.

Yes, @didiermfb this is a huge issue to me for onedir package, it requires a complete compilation system and dependencies on minion for such case :roll_eyes: 

I really don't know what to do with that  :shrug:
